src/app/space/settings/areas/create-area-dialog/create-area-dialog.component.ts

```diff
@@ -107,4 +113,13 @@ export class CreateAreaDialogComponent implements OnInit {
       });
     }
   }
+
+  get areaCreationStatus(): AreaCreationStatus {
+    if (this.errors.emptyNameFailure || this.errors.exceedLengthFailure || this.errors.uniqueValidationFailure) {
```

Comment:
Ah, I meant for the enum to look something like:

```
enum AreaCreationStatus {
  OK,
  EMPTY_NAME_FAILURE,
  EXCEED_LENGTH_FAILURE,
  UNIQUE_VALIDATION_FAILURE
}
```

and it could then replace the `errors` property. Rather than a series of mutually exclusive flags, you just use a single enum value to represent the validation state.